src/payment/strategies/adyenv2/adyenv2-payment-strategy.spec.ts

```diff
@@ -242,318 +116,337 @@ describe('AdyenV2PaymentStrategy', () => {
+            });
         });
 
-        it('returns 3DS2 IdentifyShopper flow', async () => {
-            jest.spyOn(paymentActionCreator, 'submitPayment')
-                .mockReturnValueOnce(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, identifyShopperError)))
-                .mockReturnValueOnce(submitPaymentAction);
-
-            await strategy.initialize(options);
-            await strategy.execute(getOrderRequestBody());
-
-            expect(paymentActionCreator.submitPayment).toHaveBeenCalledTimes(2);
-            expect(paymentActionCreator.submitPayment).toHaveBeenLastCalledWith({
-                methodId: 'scheme',
-                paymentData: {
-                    nonce: JSON.stringify({
-                        threeDS2Token: 'token',
-                        paymentData: 'paymentData',
+        describe('#execute', () => {
+            const identifyShopperError = getAdditionalActionError(ResultCode.IdentifyShopper);
+            const challengeShopperError = getAdditionalActionError(ResultCode.ChallengeShopper);
+            let additionalActionComponent: AdyenComponent;
+
+            beforeEach(() => {
+                let handleOnAdditionalDetails: (additionalActionState: AdyenAdditionalActionState) => {};
+
+                additionalActionComponent = {
+                    mount: jest.fn(() => {
+                        handleOnAdditionalDetails({
+                            data: {
+                                resultCode: ResultCode.ChallengeShopper,
+                                action: 'adyenAction',
+                            },
+                            isValid: true,
+                        });
+
+                        return;
                     }),
-                },
+                    unmount: jest.fn(),
+                };
+
+                jest.spyOn(adyenCheckout, 'createFromAction')
+                    .mockImplementation(jest.fn((_type, options) => {
+                        const { onAdditionalDetails } = options;
+                        handleOnAdditionalDetails = onAdditionalDetails;
+
+                        return additionalActionComponent;
+                    }));
             });
-            expect(adyenCheckout.create).toHaveBeenLastCalledWith(
-                AdyenComponentType.ThreeDS2DeviceFingerprint,
-                componentOptions
-            );
-            expect(adyenCheckout.create).toHaveBeenCalledTimes(3);
-        });
 
-        it('returns 3DS2 IdentifyShopper flow and Adyen response onError', async () => {
-            additionalActionComponent = {
-                mount: jest.fn(() => {
-                    componentOptions.onError({
-                        errorCode: 'errorCode',
-                        message: 'errorMessage',
-                    });
+            it('throws an error when payment is not present', async () => {
+                await strategy.initialize(options);
```

Comment:
it seems that you are calling `await strategy.initialize(options);` in every single test. Might it be worth moving this to `beforeEach` block of `#execute`?

Reply:
_Some_ negative tests are using different  initialization `options`  i.e.

- ` options = getInitializeOptionsWithNoCallbacks();` 

- `options = getInitializeOptionsWithUndefinedWidgetSize();`